packages/react-components/react-menu/src/components/Menu/Menu.types.ts

```diff
@@ -1,133 +1,157 @@
+     */
+    children: [JSX.Element, JSX.Element] | JSX.Element;
 
-  /**
-   * Whether the popup is open by default
-   */
-  defaultOpen?: boolean;
+    /**
+     * Whether the popup is open by default
+     *
+     * @default false
+     */
+    defaultOpen?: boolean;
 
-  /*
-   * Opens the menu on hover
-   */
-  openOnHover: boolean;
+    /**
+     * Sets the delay for mouse open/close for the popover one mouse enter/leave
+     */
+    hoverDelay?: number;
 
-  /**
-   * Opens the menu on right click (context menu), removes all other menu open interactions
-   */
-  openOnContext?: boolean;
+    /**
+     * Root menus are rendered out of DOM order on `document.body`, use this to render the menu in DOM order
+     * This option is disregarded for submenus
+     *
+     * @default false
+     */
+    inline?: boolean;
 
-  /**
-   * Root menus are rendered out of DOM order on `document.body`, use this to render the menu in DOM order
-   * This option is disregarded for submenus
-   */
-  inline?: boolean;
+    /**
+     * Call back when the component requests to change value
+     * The `open` value is used as a hint when directly controlling the component
+     */
+    onOpenChange?: (e: MenuOpenEvents, data: MenuOpenChangeData) => void;
 
-  /**
-   * Do not dismiss the menu when a menu item is clicked
-   */
-  persistOnItemClick?: boolean;
+    /**
+     * Whether the popup is open
+     *
+     * @default false
+     */
+    open?: boolean;
 
-  /**
-   * Sets the delay for mouse open/close for the popover one mouse enter/leave
-   */
-  hoverDelay?: number;
-};
+    /**
+     * Opens the menu on right click (context menu), removes all other menu open interactions
+     *
+     * @default false
+     */
+    openOnContext?: boolean;
 
-export type MenuSlots = {};
+    /**
+     * Opens the menu on hover
+     *
+     * @default false
+     */
+    openOnHover?: boolean;
 
-/**
- * Extends and drills down Menulist props to simplify API
- */
-export type MenuProps = Partial<MenuCommons> &
-  ComponentProps<MenuSlots> & {
     /**
-     * Can contain two children including {@link MenuTrigger} and {@link MenuPopover}.
-     * Alternatively can only contain {@link MenuPopover} if using a custom `target`.
+     * Do not dismiss the menu when a menu item is clicked
+     *
+     * @default false
      */
-    children: [JSX.Element, JSX.Element] | JSX.Element;
+    persistOnItemClick?: boolean;
 
     /**
      * Configures the positioned menu
      */
     positioning?: PositioningShorthand;
   };
 
-export type MenuState = MenuCommons &
-  ComponentState<MenuSlots> & {
+export type MenuState = ComponentState<MenuSlots> &
+  Pick<
+    MenuProps,
+    | 'defaultCheckedValues'
+    | 'hasCheckmarks'
+    | 'hasIcons'
+    | 'inline'
+    | 'onOpenChange'
+    | 'openOnContext'
+    | 'persistOnItemClick'
+  > &
+  Required<Pick<MenuProps, 'checkedValues' | 'onCheckedValueChange' | 'open' | 'openOnHover'>> & {
```

Comment:
This PR just 'inlines' common, but in general, why is `openOnContext` optional but `openOnHover` required?

Reply:
This is a quirk of how props are spread in state, I can make the fix so that they are both required if you want, just didn't want to meddle more with this specific PR.

Reply:
If you would like to make that fix that would be great :)